File: lua/image.lua
Image = {}
Image.__index = Image

function Image:new(width, height)
	local self = setmetatable({}, Image)
	-- this.img = imagecreate(width, height);
	self.mode = 0
	return self
end

function Image:allocateColor(r, g, b) 
	-- return imagecolorallocate($this->img, $r, $g, $b);
	return 0
end

function Image:setPixel(x, y, color)
	-- imagesetpixel($this->img, $x, $y, $color);
end

function Image:show(frame)
	if (self.mode == 2) then
		-- header('Content-Type: image/png');
		-- imagepng($this->img);
	elseif (self.mode == 3) then
		-- imagepng($this->img, sprintf("julia-%03d.png", $frame));
	elseif (self.mode == 4) then
		-- imagedestroy($this->img);
	end
end

File: js/image.js
function Image(width, height) {
	//this.img = imagecreate(width, height);
	this.mode = 0;

	this.allocateColor = function(r, g, b) {
		//return imagecolorallocate($this->img, $r, $g, $b);
		return 0
	}

	this.setPixel = function(x, y, color) {
		//imagesetpixel($this->img, $x, $y, $color);
	}

	this.show = function(frame) {
		if (this.mode == 1) {
			//header('Content-Type: image/png');
			//imagepng($this->img);
		} else if (this.mode == 2) {
			//imagepng($this->img, sprintf("julia-%03d.png", $frame));
		} else if (this.mode == 3) {
			//imagedestroy($this->img);
		}
	}
}
